### ISSUE TITLE
There was a problem while parsing the package

### ISSUE THREAD

**Issue by RukshanWithanage** [nodece/react-native-apk-installer-n#3]
When trying to install apk using the below code some devices have this issue "There was a problem while parsing the package" but some are not.

I want to find why?

* I check the APK file and it's fine isn't corrupted  
* When manually open the apk file it is fine and successfully installing without saying "There was a problem while parsing the package".

```
const filePath = RNFS.DocumentDirectoryPath + '/com.domain.example.apk';
const download = RNFS.downloadFile({
    fromUrl: 'apk file download url',
    toFile: filePath,
    progress: res => {
        console.log((res.bytesWritten / res.contentLength).toFixed(2));
    },
    progressDivider: 1
});
 
download.promise.then(result => {
    if(result.statusCode == 200){
        RNApkInstallerN.install(filePath)
    }
});
```

**Comment by nodece** [comment]
@RukshanWithanage Because I've never had this problem before, so I'm not sure what caused it. Could you capture log when installing?

**Comment by nodece** [comment]
capture reject.
```
download.promise.then(result => {
    if(result.statusCode == 200){
        RNApkInstallerN.install(filePath)
    }
}).catch(console.log);
```

**Comment by junedl-nimblechapps** [comment]
Facing the same issue.

**Comment by nodece** [comment]
I never have the issue. I can open the issue for more user.

**Comment by KrisMultiDev** [comment]
Have you found an answer to this problem because I'm having the same issue?

**Comment by nodece** [comment]
@ReaperGD I'm not sure it will help you: https://www.fixcrunch.com/mobile-problems/android-problems/there-was-a-problem-parsing-the-package/

**Comment by MarcoStb1993** [comment]
I had the same issue. For me it was necessary to allow installing APK from unknown sources for my app in android settings (using Android 10).
Go to Settings ->Apps -> YourAppName -> Install unknown apps and enable "Allow from this source". Afterwards it worked.

**Comment by arta7** [comment]
hello
after i want to install apk file after show install in error catch show error 400,now how solve this error?

**Comment by nodece** [comment]
@arta7 Backend should return 200.

**Comment by arta7** [comment]
> @arta7 Backend should return 200.

after i get status 200 in download file and show install android file,i get error and app error 400.
i use it and install  : download.promise.then(result => {
    if(result.statusCode == 200){
        RNApkInstallerN.install(filePath)
    }
}).catch(console.log);
and catch error 400 after show install page.

**Comment by nodece** [comment]
@arta7 Could you log this error? this error is provided by java.

**Comment by luandro** [comment]
Having a similar problem with Android 6. Getting this error:
```
PackageInstaller: Parse error when parsing manifest. Discontinuing installation
```

Are there known issues with Android < 7?

**Comment by sam9010** [comment]
@luandro  do you fix it?

**Comment by arta7** [comment]
hello
after i show this error,i seach in internet and i do not find any idea for
solve my problem.i change my package and it is solve my problem.
i use react-native-send-intent package and use SendIntentAndroid component
for solve my problem.

**Comment by sam9010** [comment]
@arta7 thanks a lot for your answer
can i see your example of your code? (because my problem is exactly same you problem)
i want to install a apk ( from a url in my server) ...

SendIntentAndroid.installRemoteApp("https://example.com/my-app.apk", "my-saved-app.apk").then(installWasStarted => {});
this line useful for me?

**Comment by arta7** [comment]
Yes, you can do that.

**Comment by sam9010** [comment]
i write this sample of my code but result is false
 SendIntentAndroid.installRemoteApp("https://my_url_in_my_server/sample.apk", "sample.apk").then(installWasStarted => {
            console.log("result: "+JSON.stringify(installWasStarted));
        });

**Comment by sam9010** [comment]
i do it but in my phone(android 8 samsung)  is the same error(about parsing error) :(

**Comment by arta7** [comment]
i use that and this is work for me,for control download time,i use progress
bar and state after change state i close app and android os show install
app screen.

**Comment by sam9010** [comment]
How to handle the cancel event if user press cancel button while install the apk? Thanks

**Comment by arta7** [comment]
No, it doesn't. for handle the cancel event you should change package
library in node_modules.

**Comment by sam9010** [comment]
which package?

**Comment by arta7** [comment]
> which package?

react-native-send-intent

**Comment by sam9010** [comment]
@arta7 plz go [this](https://github.com/lucasferreira/react-native-send-intent/issues/137)

**Comment by shrinarpat** [comment]
Hey, has anyone found the solution for this, I am facing the same issue

**Comment by ruanblima** [comment]
Has anyone found any solution to this problem?

**Comment by tranthientrung** [comment]
I solved this issue by add "/" in the filePath.
 const filePath = RNFS.DocumentDirectoryPath + '/' + "name.apk"
